src/spartan_app.erl

```diff
@@ -103,12 +113,9 @@ load_json_config(FileBin) ->
     lists:foreach(fun process_config_tuple/1, ConfigTuples).
 
 process_config_tuple({<<"upstream_resolvers">>, UpstreamResolvers}) ->
-    UpstreamResolverIPs = lists:map(fun parse_ipv4_address/1, UpstreamResolvers),
-    ConfigValue = [{UpstreamResolverIP, 53} || UpstreamResolverIP <- UpstreamResolverIPs],
-    application:set_env(?APP, upstream_resolvers, ConfigValue);
+    UpstreamIpsAndPorts = [parse_ipv4_address_with_port(Resolver, 53) || Resolver <- UpstreamResolvers],
```

Comment:
Rather than using a list comprehension here, can we use `lists:map(fun(Resolver) -> parse_ipv4_address_with_port(Resolver, 53) end, UpstreamResolvers)`